test/integration/node-specific/resource_clean_up.test.ts

```diff
@@ -0,0 +1,90 @@
+import { expect } from 'chai';
+
+import { runScript } from './resource_tracking_script_builder';
+
+/**
+ * This 5MB range is selected arbitrarily and should likely be raised if failures are seen intermittently.
+ *
+ * The goal here is to catch unbounded memory growth, currently runScript defaults to 100 iterations of whatever test code is passed in
+ * More than a 5MB growth in memory usage after the script has finished and _should have_ cleaned up all its resources likely indicates that
+ * the growth will continue if the script is changed to iterate the the test code more.
+ */
+const MB_PERMITTED_OFFSET = 5;
+
+describe('Driver Resources', () => {
```

Comment:
I would nest everything in here inside a `context` that explains the leak scenario being tested; this set up is specifically for MongoClient cleanup on close, so that should be clear from the logged test output when it runs